pkg/ingester/ingester.go

```diff
@@ -890,23 +897,28 @@ func (i *Ingester) PushWithCleanup(ctx context.Context, pushReq *push.Request) (
 	i.appendedSamplesStats.Inc(int64(succeededSamplesCount))
 	i.appendedExemplarsStats.Inc(int64(succeededExemplarsCount))
 
+	group := validation.GroupLabel(i.limits, userID, req.Timeseries)
+	if group != "" {
```

Comment:
I would move this check to `UpdateActiveGroupTimestamp` to avoid checking it everywhere (and state in that method's comment that it doesn't track the empty label, which is ok IMO).